### PR TITLE
Feature/#169 purge btn gi

### PR DESCRIPTION
**Pull request by d88naimi**
Purge All button added in global instance

Screen Shot 2021-03-18 at 8 03 01 AM

### CODE REVIEW

**Comment by d88naimi** [comment]
@shrunyan i'm blocked on testing this locally.  Accounts.dev invites is in a broken state, there is no way of testing user role or signing up a new user.

**Comment by shrunyan** [comment]
Have it say "Purge Instance" and add some margin so it feels like a discreet action from the instance links.

File: src/shell/components/global-instance/GlobalInstance.js [hunk 1]
```diff
@@ -100,6 +109,30 @@ export default connect(state => {
...
+                setPurge(true);
+                return request(
+                  `${CONFIG.CLOUD_FUNCTIONS_DOMAIN}/fastlyPurge?zuid=${props.instance.ZUID}`
+                ).then(res => {
+                  setPurge(false);
+                });
+              }}
+            >
+              {purge ? (
+                <FontAwesomeIcon icon={faSpinner} />
```
Comment (shrunyan) [inline src/shell/components/global-instance/GlobalInstance.js:128]: this needs the `spin` property to animate

File: src/shell/components/global-instance/GlobalInstance.js [hunk 2]
```diff
@@ -100,6 +109,30 @@ export default connect(state => {
...
+        {canPurge && (
+          <div>
+            <Button
+              kind="warn"
+              disabled={purge}
+              onClick={() => {
+                setPurge(true);
+                return request(
+                  `${CONFIG.CLOUD_FUNCTIONS_DOMAIN}/fastlyPurge?zuid=${props.instance.ZUID}`
+                ).then(res => {
```
Comment (shrunyan) [inline src/shell/components/global-instance/GlobalInstance.js:122]: use the `finally` function, so it stops on failure. add a `catch` and dispatch a notice if it failed.

**Review by shrunyan — CHANGES_REQUESTED** [review]

File: src/shell/components/global-instance/GlobalInstance.js [hunk 3]
```diff
@@ -119,17 +120,30 @@ export default connect(state => {
...
+                    });
+                  })
+                  .finally(() => {
+                    setPurge(false);
+                  });
               }}
             >
               {purge ? (
-                <FontAwesomeIcon icon={faSpinner} />
+                <FontAwesomeIcon className={styles.Spinner} icon={faSpinner} />
```
Comment (shrunyan) [inline src/shell/components/global-instance/GlobalInstance.js:142]: We do not need to style this ourselves fontawesome provides a `spin` property which handles this.

File: src/shell/components/global-instance/GlobalInstance.js [hunk 4]
```diff
@@ -119,17 +120,30 @@ export default connect(state => {
                 setPurge(true);
                 return request(
                   `${CONFIG.CLOUD_FUNCTIONS_DOMAIN}/fastlyPurge?zuid=${props.instance.ZUID}`
-                ).then(res => {
-                  setPurge(false);
-                });
+                )
+                  .then(res => {
```
Comment (shrunyan) [inline src/shell/components/global-instance/GlobalInstance.js:124]: This `then` condition is superflous as it is covered in the `finally` case

**Review by shrunyan — CHANGES_REQUESTED** [review]

**Pull request merged**